packages/flutter_tools/lib/src/android/gradle.dart

```diff
@@ -297,7 +314,17 @@ Future<String> _initializeGradle(FlutterProject project) async {
   printTrace('Using gradle from $gradle.');
   // Validates the Gradle executable by asking for its version.
   // Makes Gradle Wrapper download and install Gradle distribution, if needed.
-  await runCheckedAsync(<String>[gradle, '-v'], environment: _gradleEnv);
+  await runGradleCheckedAsync(
```

Comment:
It might be worth seeing if we can use `MemoryFileSystem` to create a fake `gradlew`. If this is the case, then we won't need to extract this method.  Caveat: I saw that `_locateGradlewExecutable` makes a call to `os.makeExecutable` which I'm not sure if it understands `MemoryFileSystem`.